File: hash-table/hash_table.cs
using System;
using System.Collections.Generic;

class HashTable {
    private List<KeyValuePair<string, int>>[] buckets;
    private int size;
    
    public HashTable(int size = 10) {
        this.size = size;
        buckets = new List<KeyValuePair<string, int>>[size];
        for (int i = 0; i < size; i++) {
            buckets[i] = new List<KeyValuePair<string, int>>();
        }
    }
    
    private int Hash(string key) {
        int sum = 0;
        foreach (char c in key) sum += c;
        return sum % size;
    }
    
    public void Put(string key, int value) {
        int index = Hash(key);
        var bucket = buckets[index];
        
        for (int i = 0; i < bucket.Count; i++) {
            if (bucket[i].Key == key) {
                bucket[i] = new KeyValuePair<string, int>(key, value);
                return;
            }
        }
        
        bucket.Add(new KeyValuePair<string, int>(key, value));
    }
    
    public int Get(string key) {
        int index = Hash(key);
        foreach (var pair in buckets[index]) {
            if (pair.Key == key) return pair.Value;
        }
        return -1;
    }
}

class Program {
    static void Main() {
        var ht = new HashTable();
        ht.Put("key1", 100);
        Console.WriteLine(ht.Get("key1"));
    }
}

File: hash-table/hash_table.go
package main
import "fmt"

type HashTable struct {
    buckets [][]Pair
    size    int
}

type Pair struct {
    key   string
    value int
}

func NewHashTable(size int) *HashTable {
    return &HashTable{
        buckets: make([][]Pair, size),
        size:    size,
    }
}

func (ht *HashTable) hash(key string) int {
    sum := 0
    for _, c := range key {
        sum += int(c)
    }
    return sum % ht.size
}

func (ht *HashTable) Put(key string, value int) {
    index := ht.hash(key)
    bucket := &ht.buckets[index]
    
    for i, pair := range *bucket {
        if pair.key == key {
            (*bucket)[i].value = value
            return
        }
    }
    *bucket = append(*bucket, Pair{key, value})
}

func (ht *HashTable) Get(key string) int {
    index := ht.hash(key)
    for _, pair := range ht.buckets[index] {
        if pair.key == key {
            return pair.value
        }
    }
    return -1
}

func main() {
    ht := NewHashTable(10)
    ht.Put("key1", 100)
    fmt.Println(ht.Get("key1"))
}